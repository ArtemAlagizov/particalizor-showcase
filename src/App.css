body {
  background: #33344c !important;
  color: white !important;
  padding: 10px;
}

#root {
  height: 100%;
}

.top-container {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.showcase-component-container {
  display: flex;
  flex-direction: column;
  margin-top: 10px;
  height: 100%;
}

.properties-container {
  display: flex;
  flex-wrap: wrap;
}

.properties-set {
  flex: 1;
}

.images {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-content: center;
  flex: 1;
}

.image-container {
  margin: 10px;
}

.image-container img {
  max-width: 100%;
}

.showcase-switch {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 5px 10px 5px 11px;
  background: #A8A8B2;
  color: black;
  cursor: pointer;
  font-weight: bold;
}

.drop-zone {
  display: flex;
  justify-content: center;
  align-items: center;
  max-width: 100%;
  min-height: 300px;
  background: #A8A8B2;
}

.image-mode-switch {
  display: flex;
  justify-content: center;
  align-self: center;
  align-items: center;
  width: 50%;
  padding: 5px 10px 5px 11px;
  background: #A8A8B2;
  color: black;
  cursor: pointer;
  font-weight: bold;
  margin-top: 10px;
  margin-bottom: 10px;
}

.dashed {
  border: 2px dashed #999;
}

.drop-image-text {
  padding: 10px;
}

.dest-canvas {
  max-width: 100%;
}