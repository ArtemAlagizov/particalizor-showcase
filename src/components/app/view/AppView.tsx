import React from "react";
import { IAppView } from "./IAppView";
import { Styled } from "../styled";
import "antd/dist/antd.css";
import MovingPictureShowcase from "../../moving-picture-showcase/container/MovingPictureShowcase";
import ParticleVortexShowcase from "../../particle-vortex-showcase/container/ParticleVortexShowcase";

const AppView: React.FC<IAppView> = ({
  switchShowcase,
  showcase,
  switchButtonTitle
}) => {
  const componentToDisplay =
    showcase === "moving-picture" ? (
      <MovingPictureShowcase />
    ) : (
      <ParticleVortexShowcase />
    );

  return (
    <Styled.TopContainerContainer>
      <Styled.ShowcaseSwitch onClick={switchShowcase}>
        {switchButtonTitle}
      </Styled.ShowcaseSwitch>
      {componentToDisplay}
    </Styled.TopContainerContainer>
  );
};

export default AppView;
