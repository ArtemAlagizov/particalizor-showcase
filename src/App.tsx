import React from "react";
import logo from "./logo.svg";
import "./App.css";
import MovingPicture from "particalizor-3000";
import "antd/dist/antd.css";
import SliderWithLabel from "./components/slider-with-label/container/SliderWithLabel";

const App: React.FC = () => {
  const onChange = () => {};
  let inputValue = 50;

  return (
    <div className="App">
      <SliderWithLabel label={"asd"} range={{ min: 0, max: 22 }} />
      <MovingPicture imageSource={logo} />
    </div>
  );
};

export default App;
