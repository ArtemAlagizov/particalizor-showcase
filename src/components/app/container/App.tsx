import React, { useState } from "react";
import AppView from "../view/AppView";

const App: React.FC = () => {
  const [selectedShowcase, setSelectedShowcase] = useState<string>(
    "moving-picture"
  );

  const switchShowcase = () => {
    setSelectedShowcase(
      selectedShowcase === "moving-picture"
        ? "particle-vortex"
        : "moving-picture"
    );
  };

  const switchButtonTitle =
    selectedShowcase === "moving-picture"
      ? "THIS IS MOVING PICTURE, CLICK TO SEE PARTICLE VORTEX"
      : "THIS IS PARTICLE VORTEX, CLICK TO SEE MOVING PICTURE";

  return (
    <AppView
      switchShowcase={switchShowcase}
      showcase={selectedShowcase}
      switchButtonTitle={switchButtonTitle}
    />
  );
};

export default App;
