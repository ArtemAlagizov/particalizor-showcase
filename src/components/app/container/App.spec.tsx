import React from "react";
import { mount, shallow } from "enzyme";
import App from "./App";
import AppView from "../view/AppView";

describe("Given a App", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(<App />);

      expect(component).toMatchSnapshot();
    });
    it("should render AppView", () => {
      const component = mount(<App />);

      expect(component.find(AppView).length).toBe(1);
    });
  });
});
