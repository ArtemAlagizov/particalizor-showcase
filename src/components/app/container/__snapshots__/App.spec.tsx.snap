// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a App when rendering should match snapshot 1`] = `
<AppView
  showcase="moving-picture"
  switchButtonTitle="THIS IS MOVING PICTURE, CLICK TO SEE PARTICLE VORTEX"
  switchShowcase={[Function]}
/>
`;
